src/overlays/actors/ovl_En_Part/z_en_part.c

```diff
@@ -21,23 +26,307 @@ const ActorInit En_Part_InitVars = {
+            this->action = 4;
+            this->actor.speedXZ = 0.0f;
+            this->actor.gravity = 0.0f;
+            this->actor.velocity.y = 0.0f;
+        }
+
+        if ((this->actor.params == 13) && (this->actor.parent != NULL) && (this->actor.parent->update == NULL)) {
+            this->actor.parent = NULL;
+        }
+    } else if (this->effectDelay <= 0) {
+        switch (this->actor.params) {
+            case 1:
+            case 9:
+            case 10:
+            case 14:
+                func_8002A6B8(globalCtx, &this->actor.posRot.pos, &zero, &zero,
+                              (s16)(this->actor.scale.y * 100.0f) * 40, 7, 0xFF, 0xFF, 0xFF, 0xFF, 0, 0xFF, 0, 1, 9, 1);
+                break;
+            case 3:
+            case 11:
+                func_8002A6B8(globalCtx, &this->actor.posRot.pos, &zero, &zero,
+                              (s16)(this->actor.scale.y * 100.0f) * 40, 7, 0xFF, 0xFF, 0xFF, 0xFF, 0, 0, 0xFF, 1, 9, 1);
+                break;
+            case 4:
+                for (i = 7; i >= 0; i--) {
+                    pos.x = this->actor.posRot.pos.x + Math_Rand_CenteredFloat(60.0f);
+                    pos.y = this->actor.posRot.pos.y + this->actor.shape.unk_08 * this->actor.scale.y +
+                            Math_Rand_CenteredFloat(50.0f);
+                    pos.z = this->actor.posRot.pos.z + Math_Rand_CenteredFloat(60.0f);
+                    velocity.y = Math_Rand_ZeroOne() + 1.0f;
+                    func_80029618(globalCtx, &pos, &velocity, &accel, Math_Rand_S16Offset(80, 100), 0x19, 0, 1);
+                }
+                break;
+            case 5:
+            case 6:
+            case 7:
+            case 8:
+                for (i = 4; i >= 0; i--) {
+                    pos.x = this->actor.posRot.pos.x + Math_Rand_CenteredFloat(25.0f);
+                    pos.y = this->actor.posRot.pos.y + Math_Rand_CenteredFloat(40.0f);
+                    pos.z = this->actor.posRot.pos.z + Math_Rand_CenteredFloat(25.0f);
+                    func_8002A6B8(globalCtx, &pos, &zero, &zero, 0x28, 7, 0xFF, 0xFF, 0xFF, 0xFF, 0, 0, 0xFF, 1, 9, 1);
+                }
+                break;
+        }
+
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    this->effectDelay--;
+    this->rotZ += this->rotZSpeed;
+}
+
+void func_80ACE5B8(EnPart* this, GlobalContext* globalCtx) {
+    this->action = 3;
+}
+
+void func_80ACE5C8(EnPart* this, GlobalContext* globalCtx) {
+    static Vec3f D_80ACF1D4 = { 0.0f, 8.0f, 0.0f };
+    static Vec3f D_80ACF1E0 = { 0.0f, -1.5f, 0.0f };
+
+    Player* player = PLAYER;
+    Vec3f velocity;
+    Vec3f accel;
+    u8 invincibilityTimer;
+
+    this->effectDelay--;
+    if (this->effectDelay == 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    velocity = D_80ACF1D4;
+    accel = D_80ACF1E0;
+
+    if (sqrt(this->actor.xyzDistFromLinkSq) <= 40.0f) {
+        invincibilityTimer = player->invincibilityTimer;
```

Comment:
Considering the usage, I suggest naming this "prev" or "old"
```suggestion
        prevInvincibilityTimer = player->invincibilityTimer;
```